event_seat_reserve/models/event_registration.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl-3.0).
+
+from odoo import fields, models
+
+
+class EventRegistration(models.Model):
+    _inherit = "event.registration"
+
+    state = fields.Selection(
+        selection_add=[("reserved", "Reserved")], ondelete={"reserved": "set default"}
+    )
+
+    def _need_pre_reservation(self):
+        return False
+
+    def action_set_reserved(self):
+        self.write({"state": "reserved"})
+        self.event_id._check_seats_availability()
+        self.event_ticket_id._check_seats_availability()
```

Comment:
Better to do this in a `write` override, like it's done in core for the "confirmed" state

https://github.com/odoo/odoo/blob/52f188849ce3a9ad9fb4e065be6cbc3fecdefb45/addons/event/models/event_registration.py#L186-L195

This will make the "constraint" have a wider effect, not only when you reserve through `action_reserve`, but also if for some reason the state field is directly changed to `reserved`.